### PR TITLE
Move symbol-cache incremental-analyzer out of oop.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Continued work to remove SolutionCrawler from OOP.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dibarbet ptal.  Thanks!

File: src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoCacheService.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.SymbolTree
+{
+    [ExportWorkspaceService(typeof(SymbolTreeInfoCacheService)), Shared]
+    internal sealed partial class SymbolTreeInfoCacheService : IWorkspaceService
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoCacheService.cs:21]: this code moved nearly verbatim from https://github.com/dotnet/roslyn/pull/63438/files#diff-31deb8089909b5074d41add123e9ce6cc85ad536dd6c4030b552cd4f55ee73a2

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoCacheService.cs:21]: (i really didn't change anything here.  feel free to review, but it's just following the logic from before).

File: src/Features/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoIncrementalAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.SolutionCrawler;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.SymbolTree
+{
+    internal partial class SymbolTreeInfoIncrementalAnalyzerProvider
+    {
+        private class SymbolTreeInfoIncrementalAnalyzer : IncrementalAnalyzerBase
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoIncrementalAnalyzer.cs:14]: this incremental-analyer now runs in proc, but shells out to OOP to perform the 3 individual steps it supports.  This is morally the same, but it does move us closer to not having solution-crawler in OOP at all.

**Review by dibarbet — APPROVED** [review]

**Review by genlu — APPROVED** [review]

**Pull request merged**